server/src/main/java/org/opensearch/search/backpressure/SearchBackpressureService.java

```diff
@@ -130,18 +167,36 @@ void doRun() {
             return;
         }
 
-        // We are only targeting in-flight cancellation of SearchShardTask for now.
-        List<SearchShardTask> searchShardTasks = getSearchShardTasks();
+        List<CancellableTask> searchTasks = getTaskByType(SearchTask.class);
+        List<CancellableTask> searchShardTasks = getTaskByType(SearchShardTask.class);
+        List<CancellableTask> cancellableTasks = new ArrayList<>();
 
         // Force-refresh usage stats of these tasks before making a cancellation decision.
+        taskResourceTrackingService.refreshResourceStats(searchTasks.toArray(new Task[0]));
         taskResourceTrackingService.refreshResourceStats(searchShardTasks.toArray(new Task[0]));
 
-        // Skip cancellation if the increase in heap usage is not due to search requests.
-        if (isHeapUsageDominatedBySearch(searchShardTasks) == false) {
+        // Check if increase in heap usage is due to SearchTasks
+        if (HeapUsageTracker.isHeapUsageDominatedBySearch(
+            searchTasks,
+            getSettings().getSearchTaskSettings().getTotalHeapPercentThreshold()
+        )) {
+            cancellableTasks.addAll(searchTasks);
+        }
+
+        // Check if increase in heap usage is due to SearchShardTasks
+        if (HeapUsageTracker.isHeapUsageDominatedBySearch(
+            searchShardTasks,
+            getSettings().getSearchShardTaskSettings().getTotalHeapPercentThreshold()
+        )) {
+            cancellableTasks.addAll(searchShardTasks);
+        }
+
```

Comment:
redundant code

Reply:
I have discussed this with @ketanv3 and decided that we will keep this for now. We can decide on removing this check in the later phase.

Reply:
Also why is domination heap based. Can it be CPU dominated as well?

Reply:
It can be and @ketanv3 is doing some rounds of testing with this. We will address this in the next PR.